cocotb_test/simulator.py

```diff
@@ -356,12 +402,12 @@ def compile_command(self):
         compile_args = self.compile_args + self.verilog_compile_args
 
         cmd_compile = (
-            ["iverilog", "-o", self.sim_file, "-D", "COCOTB_SIM=1", "-s", self.toplevel, "-g2012"]
+            ["iverilog", "-o", self.sim_file, "-D", "COCOTB_SIM=1", "-s", self.toplevel_module, "-g2012"]
             + self.get_define_commands(self.defines)
             + self.get_include_commands(self.includes)
             + self.get_parameter_commands(self.parameters)
-            + compile_args
-            + self.verilog_sources
+            + self.compile_args
```

Comment:
This is in many places:
```suggestion
            + compile_args
```